app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -52,98 +70,34 @@ private function parseReferralCode($referralCode)
             $referralCode = str_replace('iframe?', null, $referralCode);
         }
 
-        if (! empty($referralCode)) {
-            $referralCode = explode(',', $referralCode);
-
-            $referral = false;
-
-            foreach ($referralCode as $value) {
-
-                // See if we are dealing with ":" or "="
-                if (strpos($value, ':')) {
-                    $value = explode(':', $value);
-                }
-                elseif (strpos($value, '=')) {
-                    $value = explode('=', $value);
-                }
-
-                // Add northstar_id, campaign id and run, source, and source details into $values
-                switch (strtolower($value[0])) {
-                    case 'user':
-                        $values['northstar_id'] = $value[1];
-                        break;
-
-                    case 'campaignid':
-                        $values['campaign_id'] = $value[1];
-                        break;
-
-                    case 'campaign':
-                        $values['campaign_id'] = $value[1];
-                        break;
-
-                    case 'campaignrunid':
-                        $values['campaign_run_id'] = $value[1];
-                        break;
-
-                    case 'source':
-                        $values['source'] = $value[1];
-                        break;
-
-                    case 'source_details':
-                        $values['source_details'] = $value[1];
-                        break;
+        if (empty($referralCode)) {
+            return $values;
+        }
 
-                    default:
-                        break;
-                }
+        $referralCode = explode(',', $referralCode);
 
-                // Is this a referral?
-                if (strtolower($value[0]) === 'referral' && strtolower($value[1]) === 'true') {
-                    $referral = true;
-                }
```

Comment:
Reading up on the docs for [Referral Links](https://github.com/DoSomething/chompy/wiki/How-Rock-The-Vote-(RTV)-records-are-imported#special-case-referral-links), it seems we would not would want to remove checking for a `referral` query parameter. However we'll see later when looking for `$referral`...